add guide on how to install libgit2 (#13743)
kyma-project/kyma: components/function-controller/README.md

| **WEBHOOK_DEFAULTING_MINREPLICAS**        | Value of the minReplicas which webhook should set if origin equals null                   | `1`                  |
| **WEBHOOK_DEFAULTING_MAXREPLICAS**        | Value of the maxReplicas which webhook should set if origin equals null                   | `1`                  |
| **WEBHOOK_DEFAULTING_RUNTIME**            | Value of the runtime which webhook should set if origin equals null                       | `nodejs14`           |

## Troubleshooting

> ### How to build and install `libgit2` binary, required by the Function Controller, on macOS

Reviewer: Please follow the [troubleshooting template](https://github.com/kyma-project/community/blob/main/templates/resources/troubleshooting.md).

1. Navigate to the Function Controller's root directory and verify the version of `git2go`:

   ```bash
   $ cat go.mod | grep git2go

>         github.com/libgit2/git2go/v31 v31.4.14

Reviewer: Is the indentation intentional?

Author: I guess. It was like that in the [source issue](https://github.com/kyma-project/kyma/issues/13732).

   ```

> 2. Go to the git2go page to check which version of `libgit2` you must use (e.g. for `libigit2` in version 1.1 the `git2go` version is v31).

Reviewer: A link to the page would be very useful here.

> 2. Go to the git2go page to check which version of `libgit2` you must use (e.g. for `libigit2` in version 1.1 the `git2go` version is v31).

Reviewer: ```suggestion
2. Go to the git2go page to check which version of libgit2 you must use. For example, for libigit2 in version 1.1, use git2go v31.
```

Reviewer: Why the code formatting?

> 3. Clone `libgit2` repository:

Reviewer: ```suggestion
3. Clone the `libgit2` repository:
```

   ```bash
   git clone https://github.com/libgit2/libgit2.git
   ```

> 4. Checkout sources (the following example is for `git2go` version v31):

Reviewer: ```suggestion
4. Check out the sources. In this example, the sources are for git2go v31:
```

| **WEBHOOK_DEFAULTING_MINREPLICAS**        | Value of the minReplicas which webhook should set if origin equals null                   | `1`                  |
| **WEBHOOK_DEFAULTING_MAXREPLICAS**        | Value of the maxReplicas which webhook should set if origin equals null                   | `1`                  |
| **WEBHOOK_DEFAULTING_RUNTIME**            | Value of the runtime which webhook should set if origin equals null                       | `nodejs14`           |

# Troubleshooting

## How to build and install `libgit2` binary, required by the Function Controller, on macOS

### Symptom

`error: Invalid libgit2 version; this git2go supports libgit2 between vA.B.C and vX.Y.Z`

### Cause

Function Controller tests failing.

### Remedy

1. Navigate to the Function Controller's root directory and verify the version of `git2go`:

   ```bash
   $ cat go.mod | grep git2go
   ```
   You should get a result similar to this example:

   ```bash
   github.com/libgit2/git2go/v31 v31.4.14
   ```

> 2. Go to the [git2go page](https://github.com/libgit2/git2go#git2go) to check which version of libgit2 you must use. For example, for libigit2 in version 1.1, use git2go v31.

Reviewer: ```suggestion
2. Go to the [git2go page](https://github.com/libgit2/git2go#git2go) to check which version of libgit2 you must use. For example, for git2go v31, use libigit2 in version 1.1.
```

Reviewer: Shouldn't we adjust the libigit2 version to the git2go version we have?

   
3. Clone the `libgit2` repository:

   ```bash
   git clone https://github.com/libgit2/libgit2.git
   ```
4. Check out the sources. In this example, the sources are for git2go v31:
   ```bash
   git checkout v1.1.0
   ```

> 5. Build and install:

Reviewer: ```suggestion
5. Build and install the libgit2 binary:
```

| **WEBHOOK_DEFAULTING_MINREPLICAS**        | Value of the minReplicas which webhook should set if origin equals null                   | `1`                  |
| **WEBHOOK_DEFAULTING_MAXREPLICAS**        | Value of the maxReplicas which webhook should set if origin equals null                   | `1`                  |
| **WEBHOOK_DEFAULTING_RUNTIME**            | Value of the runtime which webhook should set if origin equals null                       | `nodejs14`           |

## Troubleshooting

### Symptom

Function Controller tests keep failing with such an error message:
`error: Invalid libgit2 version; this git2go supports libgit2 between vA.B.C and vX.Y.Z`

### Cause

> Function Controller tests failing due to the wrong version of the libgit2 binary. The required version of the binary is 1.1.

Reviewer: ```suggestion
Function Controller tests are failing due to the wrong version of the libgit2 binary. The required version of the binary is 1.1.
```

### Remedy

> Build and Install the libgit2 binary required by the Function Controller on macOS. Follow these steps:

Reviewer: ```suggestion
Build and install the libgit2 binary required by the Function Controller on macOS. Follow these steps:
```
